src/olympia/devhub/tests/test_tasks.py

```diff
@@ -753,60 +673,44 @@ def test_no_upgrade_annotation_no_version(self):
 
 
 class TestLegacyAddonRestrictions(ValidatorTestCase):
-    def setUp(self):
-        super(TestLegacyAddonRestrictions, self).setUp()
-
-    def test_submit_legacy_addon_restricted(self):
+    def test_legacy_submissions_disabled(self):
         file_ = get_addon_file('valid_firefox_addon.xpi')
         upload = FileUpload.objects.create(path=file_)
         tasks.validate(upload, listed=True)
 
         upload.refresh_from_db()
 
         assert upload.processed_validation['errors'] == 1
-        expected = ['validation', 'messages', 'legacy_addons_restricted']
+        expected = ['validation', 'messages', 'legacy_addons_unsupported']
         assert upload.processed_validation['messages'][0]['id'] == expected
         assert not upload.valid
 
-    def test_submit_legacy_extension_not_a_new_addon(self):
+    def test_legacy_updates_disabled(self):
         file_ = get_addon_file('valid_firefox_addon.xpi')
         addon = addon_factory(version_kw={'version': '0.1'})
         upload = FileUpload.objects.create(path=file_, addon=addon)
         tasks.validate(upload, listed=True)
 
         upload.refresh_from_db()
 
-        assert upload.processed_validation['errors'] == 0
-        assert upload.processed_validation['messages'] == []
-        assert upload.valid
-
-    def test_submit_legacy_extension_1st_version_in_that_channel(self):
-        file_ = get_addon_file('valid_firefox_addon.xpi')
-        addon = addon_factory(
-            version_kw={'version': '0.1',
-                        'channel': amo.RELEASE_CHANNEL_UNLISTED})
-        upload = FileUpload.objects.create(path=file_, addon=addon)
-        tasks.validate(upload, listed=True)
-
-        upload.refresh_from_db()
-
         assert upload.processed_validation['errors'] == 1
-        expected = ['validation', 'messages', 'legacy_addons_restricted']
+        expected = ['validation', 'messages', 'legacy_addons_unsupported']
         assert upload.processed_validation['messages'][0]['id'] == expected
         assert not upload.valid
 
-    def test_submit_legacy_extension_1st_version_in_that_channel_reverse(self):
-        file_ = get_addon_file('valid_firefox_addon.xpi')
-        addon = addon_factory(
-            version_kw={'version': '0.1',
-                        'channel': amo.RELEASE_CHANNEL_LISTED})
+    def test_submit_legacy_dictionary_disabled(self):
+        # Should not error: non-extensions types are not affected by the
+        # restriction, even if they target 57.
```

Comment:
mmm? but they should, and they do - the validation results do contain an error with the `legacy_addons_unsupported` message.

Reply:
I think I simply failed to remove that comment. Fixed in 0c2bdf06d3, nice find!